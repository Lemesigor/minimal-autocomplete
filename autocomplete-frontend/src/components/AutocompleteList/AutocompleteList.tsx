/** @jsxImportSource @emotion/react */
import {css} from '@emotion/react';
import {SuggestionItem} from '../SuggestionItem/SuggestionItem';

export type Term = {
    id: string | number;
    term: string;
};

type AutocompleteListProps = {
    items: Term[];
    handleClick: (item: Term) => void;
    isRecentSearchSuggestion: boolean;
};

export const AutocompleteList = ({
                                    items,
                                    handleClick,
                                    isRecentSearchSuggestion,
                                }: AutocompleteListProps) => {
    const dataTestId = isRecentSearchSuggestion ? 'recent-searches-list' : 'results-list';

    const listStyle = css`
        border: 1px solid #1f2937;
        border-radius: 0.375rem;
        width: 24rem;
        font-size: 1.125rem;
        background-color: white;
`;

    return (
        <ul data-testid={dataTestId} css={listStyle}>
            {items.map((item) => (
                <SuggestionItem key={item.id} item={item} handleClick={handleClick}/>
            ))}
        </ul>
    );
}
