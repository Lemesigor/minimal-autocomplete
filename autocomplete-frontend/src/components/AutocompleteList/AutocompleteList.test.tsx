/** @jsxImportSource @emotion/react */
import { render } from "@testing-library/react";
import { AutocompleteList, Term } from "./AutocompleteList";

jest.mock("../SuggestionItem/SuggestionItem", () => ({
    SuggestionItem: jest.fn(() => <li>Mocked Suggestion Item</li>),
}));

describe("AutocompleteList", () => {
    const mockItems: Term[] = [
        { id: 1, term: "apple" },
        { id: 2, term: "banana" },
    ];
    const mockHandleClick = jest.fn();

    it("should match the snapshot", () => {
        const { container } = render(
            <AutocompleteList items={mockItems} handleClick={mockHandleClick} isRecentSearchSuggestion={false} />
        );
        expect(container).toMatchSnapshot();
    });

    it("should have the correct data-testid for recent search suggestions", () => {
        const { getByTestId } = render(
            <AutocompleteList items={mockItems} handleClick={mockHandleClick} isRecentSearchSuggestion={true} />
        );
        expect(getByTestId("recent-searches-list")).toBeInTheDocument();
    });

    it("should have the correct data-testid for normal results", () => {
        const { getByTestId } = render(
            <AutocompleteList items={mockItems} handleClick={mockHandleClick} isRecentSearchSuggestion={false} />
        );
        expect(getByTestId("results-list")).toBeInTheDocument();
    });
});
