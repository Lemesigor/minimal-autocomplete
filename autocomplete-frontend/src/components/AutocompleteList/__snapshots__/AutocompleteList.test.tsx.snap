// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutocompleteList should match the snapshot 1`] = `
<div>
  <ul
    class="css-11s0iie"
    data-testid="results-list"
  >
    <li>
      Mocked Suggestion Item
    </li>
    <li>
      Mocked Suggestion Item
    </li>
  </ul>
</div>
`;
